clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -804,7 +804,33 @@ private void maybeCloseOldestConnection(long currentTimeNanos) {
     }
 
     /**
-     * Clear the results from the prior poll
+     * Clears completed receives. This is used by SocketServer to remove references to
+     * receive buffers after processing completed receives, without waiting for the next
+     * poll() after all results have been processed.
```

Comment:
Nit: maybe `after all results have been processed` is a bit redundant? Same for the `clearCompletedSends` docs.